pkg/securitypolicy/mounts.go

```diff
@@ -0,0 +1,91 @@
+//go:build linux
+// +build linux
+
+package securitypolicy
+
+import (
+	"github.com/opencontainers/runtime-spec/specs-go"
+
+	specInternal "github.com/Microsoft/hcsshim/internal/guest/spec"
+)
+
+// getDefaultMountConstraints generates mount constraints for default mounts
+// injected by containerD and/or GCS.
+func getDefaultMountConstraints(sandboxID string, spec *specs.Spec) ([]mountInternal, error) {
+	// spec.Root.Readonly is the only important field from the container spec, since
+	// it affects the "ro" mount option for networking mounts.
+	stubSpec := &specs.Spec{
+		Root: spec.Root,
+	}
+	networkingMounts := specInternal.GenerateWorkloadContainerNetworkMounts(sandboxID, stubSpec)
+	criMounts := defaultCRIMounts()
+
+	var defaultOCIMounts []specs.Mount
+	defaultOCIMounts = append(defaultOCIMounts, networkingMounts...)
+	defaultOCIMounts = append(defaultOCIMounts, criMounts...)
+
+	var defaultMountConstraints []mountInternal
+	for _, m := range defaultOCIMounts {
+		mc := newMountInternal(m.Source, m.Destination, m.Type, m.Options)
+		defaultMountConstraints = append(defaultMountConstraints, mc)
+	}
+
+	return defaultMountConstraints, nil
+}
+
+// defaultCRIMounts returns default mounts added to linux spec by containerD.
+func defaultCRIMounts() []specs.Mount {
```

Comment:
I'm wondering if we should move this somewhere else but I'm not sure where.

Reply:
`internal/guest/spec`?

Reply:
Probably nice if `pkg/securitypolicy` is just the mechanism for policy evaluation, and doesn't have any actual policy specifics in it, such as default mounts. Maybe gcs can parse the policy received from the host and then add these in as well?

Reply:
@anmaxvl is this something you're still looking at, or to be left for a future PR?

Reply:
yes, sorry, I got sidetracked a bit.

Reply:
I ended up adding a new interface, which allows extending an existing policy with some extra "global" constraints. Let me know what you think.